src/pnpq/devices/odl_thorlabs_kbd101.py

```diff
@@ -24,138 +46,202 @@ def __init__(
         self.conn.rtscts = 1
         self.resolution = 2000
         self.home_timeout = 25
-        self.move_timeout = 2
+        self.move_timeout = 4
+        self.auto_update = False
         self.maxmove = 100
         self.minmove = 0
         self.name = "Thorlabs"
         self.model = "KBD101 driver DDS100/M Stage"
+        self.logger = logging.getLogger(f"{self}")
+        self.curr_pos_steps = None
 
-    def connect(self):
-        if self.conn.is_open:
-            raise Exception("The connection is already open!")
+    def __ensure_port_open(self) -> None:
+        if not self.conn.is_open:
+            self.logger.error("disconnected")
+            raise DeviceDisconnectedError(f"{self} is disconnected")
+
+    def __ensure_device_in_range(self, steps: int) -> None:
+        max_threshold = self.maxmove * self.resolution
+        min_threshold = self.minmove * self.resolution
+
+        if steps > max_threshold or steps < min_threshold:
+            raise OdlMoveOutofRangeError(
+                f"Move request for device{self} is out of range min({self.minmove}) - max({self.maxmove})"
+            )
 
+    def connect(self) -> None:
+        if self.conn.is_open:
+            self.logger.warning("ODL ({self}) serial connection is already open")
         self.conn.open()
-        print("Connecting to Thorlabs ODL mdoule")
-        # Enable Channel ID (0)
-        self.conn.write(b"\x10\x02\x01\x01\x50\x01")
-        time.sleep(0.5)
 
-    def identify(self):
         if not self.conn.is_open:
-            raise RuntimeError("Identification failed: can not connect to odl device")
+            raise DeviceDisconnectedError(f"ODL device is disconneced")
+        self.logger.info(f"({self}): Connecting to Thorlabs ODL module")
+        # Enable Channel ID (0)
+        self.conn.write(ENABLE_CHANNEL_SET_COMMAND)
+        time.sleep(0.1)
+        self.conn.write(ENABLE_CHANNEL_GET_COMMAND)
+        en_channel_result = self.__waitForReply(b"\x12\x02", 5)
+        if en_channel_result is None:
+            self.logger.error(f"can not enable odl channel")
 
-        self.conn.write(b"\x23\x02\x00\x00\x50\x01")
+    def identify(self):
+        self.__ensure_port_open()
+        self.conn.write(ODL_IDENTIFY_COMMAND)
 
-    def waitForReply(self, sequence, timeout):
+    def __waitForReply(self, sequence, timeout):
```

Comment:
should be `__wait_for_reply`.
stick to snake case for readability and easy to expect.
imagine you are writing code, if the methods named `__wait_for_reply` and `__waitForReply` exist in the codebase, you will typo and loose your time. once you decided to use snake_case for the method names, you must stick to it.